tests/pipelines/test_pipelines_common.py

```diff
@@ -2153,6 +2157,41 @@ def test_StableDiffusionMixin_component(self):
             )
         )
 
+    @require_torch_gpu
+    @slow
+    def test_torch_compile_recompilation(self):
+        inputs = self.get_dummy_inputs(torch_device)
+        components = self.get_dummy_components()
+
+        pipe = self.pipeline_class(**components).to(torch_device)
+        if getattr(pipe, "unet", None) is not None:
+            pipe.unet = torch.compile(pipe.unet, fullgraph=True)
+        else:
+            pipe.transformer = torch.compile(pipe.transformer, fullgraph=True)
+
+        with torch._dynamo.config.patch(error_on_recompile=True):
+            _ = pipe(**inputs)
```

Comment:
This is good. Another alternative that does not require patching is

```
torch._dynamo.config.recompile_limit = True
```

That with fullgraph=True will ensure that that torch.compile will raise an error on recompilation.